pkg/payload/payload.go

```diff
@@ -224,6 +224,11 @@ func shouldExclude(excludeIdentifier, profile string, manifest *manifest.Manifes
 		return true
 	}
 
+	techPreviewAnnotation := "release.openshift.io/feature-gate"
+	if annotations[techPreviewAnnotation] == "TechPreviewNoUpgrade" && !includeTechPreview {
```

Comment:
nit: The annotation _key_ is generically about feature gates.  The only tech-preview-ness that comes in is the value.  So `featureGateAnnotation`?  Or drop the local variable and just inline the key literal in the `if` line?

Reply:
Also, [from the enhancement][1]:

> If the value is not "TechPreviewNoUpgrade", the manifest is never created.

So:

```golang
if featureGate, ok := annotations["release.openshift.io/feature-gate"]; ok && !(includeTechPreview && featureGate == configv1.TechPreviewNoUpgrade) {
  return true
}
```

or something?

[1]: https://github.com/openshift/enhancements/pull/957/files#diff-6f9f82c1e111102d550554343fc634b0908acf22a0aa1490c0801f7ddf85b213R61

Reply:
fixed in fda77fe; thanks :)